Sources/SourceKitLSP/Clang/ClangLanguageServer.swift

```diff
@@ -32,6 +32,27 @@ extension NSLock {
   }
 }
 
+/// Gathers data from clangd's stderr pipe. When it has accumulated a full line, writes the the line to the logger.
+fileprivate class ClangdStderrLogForwarder {
+  private var buffer = Data()
+
+  func handle(_ newData: Data) {
+    self.buffer += newData
+    if newData.contains(UInt8(ascii: "\n")) {
+      self.flushBuffer()
+    }
+  }
+
+  private func flushBuffer() {
+    guard let newlineIndex = self.buffer.firstIndex(of: UInt8(ascii: "\n")) else {
+      return
+    }
```

Comment:
Would it be better to use `lastIndex` here? I guess it doesn't happen with the current implementation since we're reading a byte at a time, but if we weren't and e.g if multiple lines got read at once, presumably we'd want to forward all of them here. Plus the newline is more likely to be nearer the end than the start in general.

Reply:
The quest is whether we want a separate os_log message for every line from clangd or if we are fine with merging multiple clangd stderr lines into a single os_log message. 

I don’t have super strong opinions here but one thing to keep in mind is that os_log has a limit to how long the log message can be (something around 1KB). So, if we merge too many lines together os_log might truncate. That’s why I chose to only have one line per os_log message, with the assumption that usually a single line will be less than 1024 characters long.

Reply:
Makes sense, my only real concern here is accidentally ending up in a situation where we don't fully flush the buffer. If we do switch to `availableData`, should we be making an effort to split lines here?

Reply:
I changed it to a `while` loop so that we flush out all the lines to separate log messages and added a comment explaining why every line is its own message.